brave/brave-core
# Make ConnectHW screen working for both Trezor/Ledger devices

Resolves https://github.com/brave/brave-browser/issues/19495
Resolves https://github.com/brave/brave-browser/issues/19485

- Added screen to Ledger/Trezor messages and transaction signing operations
- Added cancel event to keyrings to reset state and close popups if user rejected operation
- Reworked statuses indication for ConfirmationView
- Added Ledger device disconnect detection and update the panel status accordingly
- Added new status to open correct app:
![image](https://user-images.githubusercontent.com/2965009/142862029-39a4dd23-aabc-401c-9d9a-304b8a740c14.png)


## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Sign transactions and messages by Ledger/Trezor
- Make sure it shows connect HW screen with correct statuses
- Disconnect, connect, device in different state during transaction confirmation, all statuses should be visible on screen
- Try to enable not Ethereum Apps on Ledger device, the panel should detect it and ask to enable Ethereum app
